plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -263,12 +284,18 @@ public static ReaderPageSource createPageSource(
                 .orElse(columns);
 
         for (HiveColumnHandle column : baseColumns) {
-            checkArgument(column.getColumnType() == REGULAR, "column type must be REGULAR: %s", column);
+            checkArgument(column == ROW_INDEX_COLUMN || column.getColumnType() == REGULAR, "column type must be REGULAR: %s", column);
+        }
+        if (columns.indexOf(ROW_INDEX_COLUMN) != columns.lastIndexOf(ROW_INDEX_COLUMN)) {
+            throw new IllegalArgumentException("ROW_INDEX_COLUMN may only be requested once");
         }
 
         ImmutableList.Builder<Type> trinoTypes = ImmutableList.builder();
         ImmutableList.Builder<Optional<Field>> internalFields = ImmutableList.builder();
         for (HiveColumnHandle column : baseColumns) {
+            if (column == ROW_INDEX_COLUMN) {
+                continue; // The index column is not passed to the PageSource
```

Comment:
remove if you follow the

> I'd consider fusing RowIndexPageSource with ParquetPageSource, 


comment

Reply:
This might be a good argument in favor of merging `RowIndexPageSource` into `ParquetPageSource`, actually. Then the `PageSourceFactory` doesn't need to do as much sanity-checking, which feels cleaner.